wapitiCore/attack/mod_http_headers.py

```diff
@@ -0,0 +1,78 @@
+import requests
+from wapitiCore.attack.attack import Attack
+from wapitiCore.net.web import Request
+from wapitiCore.language.vulnerability import Additional
+#This module check the recommendations of security parameters in HTTP headers
+class mod_http_headers(Attack):
+
+    name = "http_headers"
+    check_list = ['DENY', 'SAMEORIGIN', 'ALLOW-FROM']
+
+    def is_set(self, request: object, header_name, check_list):
+        if header_name == "X-Frame-Options":
+            if 'X-Frame-Options' not in request.headers:
+                return False
+            else:
+                return any(element in request.headers['X-Frame-Options'] for element in check_list)
+        if header_name == "X-XSS-Protection":
+            if "X-XSS-Protection" not in request.headers:
+                return False
+            else:
+                return '1' in request.headers['X-XSS-Protection']
+        if header_name == "X-Content-Type-Options":
+            if "X-Content-Type-Options" not in request.headers:
+                return False
+            else:
+                return "nosniff" in request.headers['X-Content-Type-Options']
+        if header_name =="Strict-Transport-Security":
+            if 'Strict-Transport-Security' not in request.headers:
+                return False
+            else:
+                return "max-age=" in request.headers['Strict-Transport-Security']
+
+    def attack(self):
+        url = self.persister.get_root_url()
+        session = requests.session()
+        req = session.get(url)
+        request = Request(url)
+        self.log_blue("Checking X-Frame-Options :")
```

Comment:
You should remove colons otherwise it seems weird when nothing is found  (looks like something is missing) : 

```
./bin/wapiti -u https://portswigger.net/ --scope url -v2 -m http_headers

     __      __               .__  __  .__________
    /  \    /  \_____  ______ |__|/  |_|__\_____  \ 
    \   \/\/   /\__  \ \____ \|  \   __\  | _(__  <
     \        /  / __ \|  |_> >  ||  | |  |/       \ 
      \__/\  /  (____  /   __/|__||__| |__/______  /
           \/        \/|__|                      \/
Wapiti-3.0.3 (wapiti.sourceforge.io)
[+] GET https://portswigger.net/ (0)
[*] Enregistrement de l'état du scan, veuillez patienter...

 Note
========
Ce scan a été sauvé dans le fichier /home/devloop/.wapiti/scans/portswigger.net_url_48a448c8.db
[*] Wapiti a trouvé 1 URLs et formulaires lors du scan
[*] Chargement des modules :
         backup, blindsql, buster, crlf, delay, exec, file, htaccess, http_headers, methods, nikto, permanentxss, redirect, shellshock, sql, ssrf, wapp, xss, xxe

[*] Lancement du module http_headers
Checking X-Frame-Options :
Checking X-XSS-Protection
Checking X-Content-Type-Options :
Checking Strict-Transport-Security :
```

Reply:
Fixed